### ISSUE TITLE
Deploying binary functions individually fails

### ISSUE THREAD

**Issue by siegesmund** [serverless/serverless-openwhisk#79]
`sls deploy -f foo`  fails with a compiled binary. `sls deploy` deploys all functions, including the binary, successfully. 

It fails at `serverless-openwhisk/compile/functions/runtimes/base.js`.

Specifically, in 

`getArtifactPath(functionObject) {
    return this.serverless.service.package.individually ? 
      functionObject.artifact : this.serverless.service.package.artifact;
  }`, 

functionObject.artifact is undefined, causing getArtifactZip to fail and crash the deployment.
 [In the Serverless Framework, the artifact appears to be stored at functionObject.package.artifact.](https://github.com/serverless/serverless/blob/ed3c7c371833ef82bdef21fa4e5dbca0100cd2f5/lib/plugins/package/lib/packageService.js#L44)

Changing `functionObject.artifact` to `functionObject.package.artifact` in the Openwhisk plugin allows the function to be deployed.

**Comment by jthomas** [comment]
Hey @siegesmund, can you share your `serverless.yml`?

**Comment by siegesmund** [comment]
This is an example. One deviation from your example functions is that runtimes are defined on a per-function basis, something that works with other cloud provider plugins.

`
service: foobar

provider:
  name: openwhisk

functions:
  # Returns golang binary version
  version:
    runtime: binary
    handler: functions/version/main

plugins:
  - serverless-openwhisk
`

To get around the error, I've forked and modified the serverless-openwhisk plugin and am testing it locally. So far, it deploys both binary and node.js functions both individually and on a per-service basis. I haven't tried Swift, Python, etc.

**Comment by siegesmund** [comment]
I have created a [repo](https://github.com/siegesmund/openwhisk_test) which recreates the problem on my machine. Again, the problem is:

sls deploy             >> works
sis deploy -f test  >> fails

I encountered this bug while writing an [openwhisk go plugin](https://github.com/siegesmund/serverless-openwhisk-go) which allows a developer to manage the full Go action lifecycle from the sls command line, a significant workflow improvement that already exists for AWS and Azure. I haven't published it yet, however, pending some clarity on this issue.
